MekHQ/src/mekhq/campaign/stratcon/StratconTrackState.java

```diff
@@ -148,12 +149,28 @@ public void updateScenario(StratconScenario scenario) {
         }
     }
 
+    public void removeScenario(int campaignScenarioID) {
+        if (getBackingScenariosMap().containsKey(campaignScenarioID)) {
+            removeScenario(getBackingScenariosMap().get(campaignScenarioID));
+        }
+    }
+    
     /**
      * Removes a StratconScenario from this track.
      */
     public void removeScenario(StratconScenario scenario) {
         scenarios.remove(scenario.getCoords());
         getBackingScenariosMap().remove(scenario.getBackingScenarioID());
+        Map<StratconCoords, StratconStrategicObjective> objectives = getObjectivesByCoords();
+        if (objectives.containsKey(scenario.getCoords())) {
+            StrategicObjectiveType objectiveType = objectives.get(scenario.getCoords()).getObjectiveType();
+            
+            switch (objectiveType) {
+                case RequiredScenarioVictory:
+                case SpecificScenarioVictory:
+                    objectives.remove(scenario.getCoords());
+            }
```

Comment:
```suggestion
                case SpecificScenarioVictory:
                    objectives.remove(scenario.getCoords());
                    break;
                default:
                    break;
            }
```
This fixes the missing cases warning, provided the missing cases are not required here.

Reply:
A little redundant, but ok.